src/ctia/http/middleware/cache_control.clj

```diff
@@ -0,0 +1,33 @@
+(ns ctia.http.middleware.cache-control
+  (:require [ctia.lib.time :refer [format-rfc822-time]]
+            [pandect.algo.sha1 :refer [sha1]])
+  (:import java.io.File))
+
+(defn- read-request? [request]
+  (#{:get :head} (:request-method request)))
+
+(defn- ok-response? [response]
+  (= (:status response) 200))
+
+(defn calculate-etag [body]
+  (case (class body)
+    String (sha1 body)
+    File (str (.lastModified body) "-" (.length body))
+    (sha1 (.getBytes (pr-str body) "UTF-8"))))
+
+(defn update-headers [headers etag body]
+  (merge headers
```

Comment:
You could use cond-> with assoc instead of merge (it makes the when-let below a little cleaner, IMO).  For the "ETag" key, you could assoc it under the :else predicate.  Just a nit, though.  :)